### PR TITLE
Update selection changing events

### LINKED ISSUE #13595: igx-combo selectionChanging has inconsistent arguments depending on valueKey
## Description  
If `valueKey` is set, then the argument type for `newSelection`, `oldSelection`, etc. of the combo's selectionChanging event arguments is different from the type if `valueKey` is not set. 

 * igniteui-angular version: 16.1.x
 * browser: all

## Steps to reproduce  

1. Create an `igx-combo`.
2. Bind it to data.
3. Bind to `selectionChanging`
4. Set a `valueKey`
5. Observe `selectionChanging`
6. Remove the `valueKey`
7. Observe `selectionChanging`

## Result  
The arguments differ.

## Expected result  
The event arguments should be consistent.

### PR DESCRIPTION
**Pull request by wnvko**
Change the `oldSelection` and `newSelection` properties of both combos selection changing arguments to `oldValue` and `newValue`. Also add `oldSelection` and `newSelection` which should always emit the entire selected items.

Also fixes disappearing combo items in the input string when combo is bound to remote data.

Closes #13595  

### Additional information (check all that apply):
 - [x] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [x] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [x] This PR contains breaking changes
 - [x] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by kdinev — APPROVED** [review]

**Review by kdinev — DISMISSED** [review]

**Review by damyanpetev — CHANGES_REQUESTED** [review]

**Comment by valadzhov** [comment]
This is the result when setting the `valueKey` to `null` and this might be a minor issue:
image

File: projects/igniteui-angular/src/lib/combo/combo.common.ts [hunk 1]
```diff
@@ -1295,11 +1295,13 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
...
-        return keys.map(key => isNaNvalue(key)
-            ? this.data.find(entry => isNaNvalue(entry[this.valueKey]))
-            : this.data.find(entry => entry[this.valueKey] === key))
-            .filter(e => e !== undefined);
+        return keys.map(key => {
+            const item = isNaNvalue(key)
+                ? this.data.find(entry => isNaNvalue(entry[this.valueKey]))
+                : this.data.find(entry => entry[this.valueKey] === key);
+
+            return item !== undefined ? item : { [this.valueKey]: key };
```
[the remark this answers is not in the document]
Comment (wnvko) [reply projects/igniteui-angular/src/lib/combo/combo.common.ts:1303]: done

**Review by damyanpetev — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.spec.ts [hunk 2]
```diff
@@ -1799,7 +1807,7 @@ describe('IgxSimpleCombo', () => {
 
                 expect(combo.displayValue).toEqual([]);
                 expect(combo.selection).toEqual([]);
-                expect(combo.value).toEqual([]);
+                expect(combo.value).toEqual(['']);
```
[the remark this answers is not in the document]
Comment (wnvko) [reply projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.spec.ts]: this test makes no sense, but is fixed now.
Single combo with array value...

**Comment by wnvko** [comment]
> This is the result when setting the `valueKey` to `null` and this might be a minor issue: image

this should not be supported.

**Comment by ChronosSF** [comment]
@wnvko , please fix the conflict

**Review by kdinev — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.spec.ts [hunk 3]
```diff
@@ -1786,17 +1794,21 @@ describe('IgxSimpleCombo', () => {
...
+                    { field: 'empty string', value: [''] },
+                    { field: 'null', value: [null] },
+                    { field: 'NaN', value: [NaN] },
+                    { field: 'undefined', value: [undefined] },
                 ];
 
+                fixture.componentInstance.reactiveForm.resetForm();
+                fixture.detectChanges();
+
                 expect(combo.displayValue).toEqual([]);
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.spec.ts:1812]: 👀 I see we have our work cut out for us

File: projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.spec.ts [hunk 4]
```diff
@@ -1786,17 +1794,21 @@ describe('IgxSimpleCombo', () => {
                 fixture.detectChanges();
                 combo = fixture.componentInstance.reactiveCombo;
             });
-            it('should not select null, undefined and empty string in a reactive form with required', () => {
+            // TODO: this test makes no sense - reactive combo with values equal to null, undefined and empty string
+            it('should not select null, undefined and empty string in a reactive form with required', fakeAsync(() => {
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.spec.ts:1798]: Excluding the fact the `value` getter is weird, the form control integration actually works (or should) correctly and should be fed single items. Also the test is trying to _not_ select some values so the required validator triggers, no?

**Review by damyanpetev — APPROVED** [review]

**Pull request merged**